File: test56_out/test56/TestProg/TestProg.c

/*
 * ======== string literals ========
 */

static const char test56_TestProg__s__fileName[] = "TestProg.p";
static const char test56_TestProg__s__6[] = "7. Protocol function type for function ref Array, with arg=array index.\n";
static const char test56_TestProg__s__1[] = "2. Init on1_fref with assignment, then use it to call On1.on(), with arg=8.\n";
static const char test56_TestProg__s__3[] = "4. Init on2_fref with assignment, then use it to call On3.on(), with arg=7.\n";
static const char test56_TestProg__s__8[] = "9. Array of function refs init from function names, with arg=array index.\n";
static const char test56_TestProg__s__9[] = "TestProg";
static const char test56_TestProg__s__7[] = "8. Array of function refs init from function names, with arg=array index.\n";
static const char test56_TestProg__s__4[] = "5. Array of fcn refs: target array, target fcnrefs, target fcns, with arg=array index.\n";
static const char test56_TestProg__s__0[] = "1. Init on1_0_fref with assignment, then use it to call On1.on_0(), no args.\n";
static const char test56_TestProg__s__5[] = "6. Array of fcn refs: host array, host fcnrefs, target fcns, with arg=array index.\n";
static const char test56_TestProg__s__2[] = "3. Init on2_fref on dcln, then use it to call On2.on(), with arg=3.\n";

/*
 * ======== code injections ========
 */

 #include <stdlib.h> 

/*
 * ======== function definitions ========
 */

void test56_TestProg_pollen__run__E() {
    uint8 i;
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__0);
    (test56_TestProg_on1_0_fref__V) = (test56_On1_on_0__E);
    test56_TestProg_on1_0_fref__V();
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__1);
    (test56_TestProg_on1_fref__V) = (test56_On1_on__E);
    test56_TestProg_on1_fref__V(8);
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__2);
    test56_TestProg_on2_fref__V(3);
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__3);
    (test56_TestProg_on2_fref__V) = (test56_On3_on__E);
    test56_TestProg_on2_fref__V(7);
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__4);
    for (i = 0; (i) < 3; (i)++) {
        (*test56_TestProg_on_functions__A[(i)])(i);
    }
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__5);
    for ((i) = 0; (i) < 3; (i)++) {
        (*test56_TestProg_on_host_functions__A[(i)])(i);
    }
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__6);
    for ((i) = 0; (i) < 3; (i)++) {
        (*test56_TestProg_on_functions_proto__A[(i)])(i);
    }
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__7);
    for ((i) = 0; (i) < 4; (i)++) {
        (*test56_TestProg_on_function_by_name__A[(i)])(i);
    }
    test56_TestProg_pollen__printStr((string)test56_TestProg__s__8);
    (test56_TestProg_on_function_by_name__A[3]) = (test56_On5_on__E);
    for ((i) = 0; (i) < 4; (i)++) {
        (*test56_TestProg_on_function_by_name__A[(i)])(i);
    }
     exit(0) ;

}

void test56_TestProg_targetInit__I() {
}

File: test56_out/test56/TestProg/TestProg.js

/*
 * ======== MODULE test56.TestProg ========
 */

var $$c = function() {
    const TestProg = {};
    TestProg.TestProg = TestProg
    TestProg.$name = 'test56.TestProg';
    TestProg.pollen$used = 0;
    TestProg.On1 = $units['test56.On1'];
    TestProg.On2 = $units['test56.On2'];
    TestProg.On3 = $units['test56.On3'];
    TestProg.On4 = $units['test56.On4'];
    TestProg.On5 = $units['test56.On5'];
    TestProg.PrintImpl = $units['test56.PrintImpl'];
    TestProg.pollen__run = new $$Ref('test56_TestProg_pollen__run__E');
    TestProg.targetInit = new $$Ref('test56_TestProg_targetInit__I');
    TestProg.$$hostInit = function() {
        var $$text = '';
        TestProg.on_host_functions = new $$Array(3, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_host_functions__A', false); TestProg.on_host_functions.$$assign((new $$Array(3, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ (TestProg.on3_host_fref), (TestProg.on3_host_fref), (TestProg.on3_host_fref) ]));
        TestProg.on_functions = new $$Array(3, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_functions__A', false); TestProg.on_functions.$$assign((new $$Array(3, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ (TestProg.on1_fref), (TestProg.on2_fref), (TestProg.on3_fref) ]));
        TestProg.on_functions_proto = new $$Array(3, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_functions_proto__A', false); TestProg.on_functions_proto.$$assign((new $$Array(3, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ (TestProg.on1_fref), (TestProg.on2_fref), (TestProg.on3_fref) ]));
        TestProg.on_function_by_name = new $$Array(4, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_function_by_name__A', false); TestProg.on_function_by_name.$$assign((new $$Array(4, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ ($units['test56.On1'].on), ($units['test56.On2'].on), ($units['test56.On3'].on), ($units['test56.On4'].on) ]));
        return $$text;
    }
    TestProg.pollen__uses__ = function() {
        $$bind($units['test56.On1'], 'pollen$used', true);
        $$bind($units['test56.On2'], 'pollen$used', true);
        $$bind($units['test56.On3'], 'pollen$used', true);
        $$bind($units['test56.On4'], 'pollen$used', true);
        $$bind($units['test56.On5'], 'pollen$used', true);
        $$bind($units['test56.PrintImpl'], 'pollen$used', true);
    }
    TestProg.$$privateInit = function() {
        TestProg.on1_fref = new $$Ref('test56_On1_on__E');
        TestProg.on1_0_fref = new $$Ref('test56_On1_on_0__E');
        TestProg.on2_fref = new $$Ref('test56_On2_on__E');
        TestProg.on3_fref = new $$Ref('test56_On3_on__E');
        TestProg.on1_0_hfref = new $$Ref('test56_On1_on_0__E');
        TestProg.on3_host_fref = new $$Ref('test56_On3_on__E');
        TestProg.on1_0_fref_4 = new $$Ref('test56_On1_on_0__E');
var $$v = 'test56_PrintImpl';
        var $$s = $$bind($units['test56.TestProg'], 'pollenPrintProxy', $$v);{ $$v.pollen$used = true;
        $units['test56.PrintImpl'].pollen$used = true;}

        TestProg.on_functions = new $$Array(3, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_functions__A', false); TestProg.on_functions.$$assign((new $$Array(3, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ (TestProg.on1_fref), (TestProg.on2_fref), (TestProg.on3_fref) ]));
        TestProg.on_functions_proto = new $$Array(3, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_functions_proto__A', false); TestProg.on_functions_proto.$$assign((new $$Array(3, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ (TestProg.on1_fref), (TestProg.on2_fref), (TestProg.on3_fref) ]));
        TestProg.on_function_by_name = new $$Array(4, function($$cn,$$idx){ return 0;}, 'test56_TestProg_on_function_by_name__A', false); TestProg.on_function_by_name.$$assign((new $$Array(4, function($$cn,$$idx){ return 0;}, '__new_'+($$Array.$$idx++)+'__A', false)).$$assign([ ($units['test56.On1'].on), ($units['test56.On2'].on), ($units['test56.On3'].on), ($units['test56.On4'].on) ]));
    }
    return TestProg;
}
var $$u = $$c();
$units['test56.TestProg'] = $$u;
$units.push($$u);

